src/main/java/rx/internal/operators/OperatorObserveOn.java

```diff
@@ -182,9 +176,9 @@ protected void schedule() {
         void pollQueue() {
             int emitted = 0;
             do {
-                counter = 1;
+                counter.set(1);
                 long produced = 0;
-                long r = requested;
+                long r = requested.get();
```

Comment:
Performance. `counter.set(1)` is a StoreLoad barrier that forces the compiler to re-read `requested` from the instance field which may take 1-3 ns per iteration. Keeping it in a local variable thus likely in a register is only 0.5 - 1 ns on my i7 4790 desktop. Later on, the `counter.decrementAndGet` acts also a StoreLoad barrier and will force the `counter` reference to be re-read. These were nonexistent overheads in the original version.